src/ImageSharp/Formats/Pbm/BinaryDecoder.cs

```diff
@@ -162,23 +161,12 @@ private static void ProcessBlackAndWhite<TPixel>(Configuration configuration, Bu
             for (int x = 0; x < width;)
             {
                 int raw = stream.ReadByte();
-                int bit = startBit;
-                startBit = 0;
-                for (; bit < 8; bit++)
+                int stopBit = Math.Min(8, width - x);
```

Comment:
Should be a nice perf boost here also.

Reply:
Indeed, solving a bug by simplifying the code